### PR TITLE
Improve install.sh

### LINKED ISSUE #153: [Feature] Uninstall Script
**Is your feature request related to a problem? Please describe.**

So far, it is very complex to correctly uninstall all the liqo resources.

**Describe the solution you'd like**

Similarly to install.sh script, we should have a uninstall.sh script to safely remove all liqo objects.

**Describe alternatives you've considered**

Liqo uninstall is uncomfortable and requires to have helminstalled and liqo repository downloaded.

Steps:

- [x] Implement Uninstall function (Ref. #185)
- [x] Linting Shellcheck (Ref. #229)
- [x] Sleep deletion + Auto-crt approval (Ref. #232)
- [ ] Default mode: Install the latest "stable" release
- [x] Checkout tagged releases

### PR DESCRIPTION
**Pull request by giorio94**
# Description

This PR refactors and improves the `install.sh` script to:
* Automatically download the latest tagged version (any version/commit or master can be manually specified through the `LIQO_VERSION` variable)
* Automatically retrieve the latest commit from master, to avoid `latest` image tags
* Checks whether the requested Liqo version does exist before trying to download it
* Improve the installation logs, and make them more pleasant (e.g. Liqo logo and colors, as in Oh-My-Zsh)
* Major refactor the script, making it more structured and (hopefully) look more professional
* Support both wget and curl to download files
* Add a bunch of unit tests to ensure the correct functioning of the installer (complemented by the E2E tests)

Fixes #153 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [X] Manual testing
- [X] New unit tests
- [x] E2E tests

### CODE REVIEW

**Comment by giorio94** [comment]
/hold

Some modifications are still required before merging (e.g. slight adaptations to the documentation and, maybe, to the CI).
However, I would like to know your feedback before going on with that.

**Comment by palexster** [comment]
> /hold
> 
> Some modifications are still required before merging (e.g. slight adaptations to the documentation and, maybe, to the CI).
> However, I would like to know your feedback before going on with that.

I think this PR is a massive step forward.

**Review by palexster — DISMISSED** [review]

File: install.sh
Comment (palexster) [inline install.sh:387]: A question for @aleoli: Do you have a list of all conditions that should be waited to move on?

Comment (aleoli) [reply install.sh:387]: No, at the moment there are no network-related info in ForeignClusters

File: install.sh
Comment (AbakusW) [inline install.sh:420]: small typo (**uninstaled**") here and in purge_liqo()

Comment (giorio94) [reply install.sh:420]: Thanks!

**Comment by giorio94** [comment]
All tests passed, hence should be ready for review

**Warning**: before merging, it is necessary to adapt the liqops pipeline and drop the second commit

**Review by palexster — APPROVED** [review]

**Review by mLavacca — APPROVED** [review]

**Pull request merged**
